From linux-usb-owner@vger.kernel.org  Thu Aug 19 10:37:00 2010
Message-Id: <201008102129.o7ALTJ3b017160@imap1.linux-foundation.org>
Subject: USB: serial: io_ti.c: don't return 0 if writing the download record failed
To: greg@kroah.com
Cc: linux-usb@vger.kernel.org, akpm@linux-foundation.org,
	roel.kluin@gmail.com
From: Roel Kluin <roel.kluin@gmail.com>
Date: Tue, 10 Aug 2010 14:29:19 -0700

From: Roel Kluin <roel.kluin@gmail.com>

If the write download record failed we shouldn't return 0.

Signed-off-by: Roel Kluin <roel.kluin@gmail.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/serial/io_ti.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/usb/serial/io_ti.c
+++ b/drivers/usb/serial/io_ti.c
@@ -1284,7 +1284,7 @@ static int download_fw(struct edgeport_s
 				kfree(header);
 				kfree(rom_desc);
 				kfree(ti_manuf_desc);
-				return status;
+				return -EINVAL;
 			}
 
 			/* Update I2C with type 0xf2 record with correct
